### PR TITLE
StreamArray Slicing

### LINKED ISSUE #4: TODO: Add Slicing Support to StreamArray
# TODO: Add Slicing Support to StreamArray

---

## Description

The StreamArray object should support slicing to allow users to easily retrieve subsets of streams. This enhancement will improve the usability of the object, enabling developers to work more flexibly with collections of streams.
> [!NOTE]
> The StreamArray object already supports indexing all you habe to do is impliment slicing.

---

Scope of Work

1. Add Slicing Functionality

    Implement the `__getitem__` method in StreamArray to support slicing.

    Return a new StreamArray instance containing the sliced streams.

2. Write Unit Tests

    Add tests for:

    - Valid slicing cases (e.g., stream_array[1:4], stream_array[:2]).

    - Negative slicing (e.g., stream_array[-3:]).

    - Step slicing (e.g., stream_array[::2]).

    - Invalid slicing cases (e.g., non-integer indices).

3. Handle Edge Cases

    - Ensure empty slices return an empty StreamArray.

    - Handle invalid inputs gracefully (e.g., raise TypeError for non-slice inputs).

4. Update Documentation(Optional)

    Document how to use slicing with StreamArray, including examples.

---

## Proposed Implementation

Code Snippet
```py
class StreamArray:
    def __init__(self, streams):
        self._streams = streams

    def __getitem__(self, index):
        if isinstance(index, slice):
            return StreamArray(self._streams[index])  # Return a new instance for slicing
        elif isinstance(index, int):
            return self._streams[index]  # Return a single stream
        else:
            raise TypeError("Index must be an int or a slice.")

    def __len__(self):
        return len(self._streams)

    def __repr__(self):
        return f"StreamArray({self._streams})"
```

---

## Acceptance Criteria

1. The StreamArray object supports slicing seamlessly.

2. Unit tests cover all relevant scenarios and edge cases.

3. Documentation includes examples and explanations for slicing.

---

## Labels

enhancement, help wanted, testing, documentation, good first issue

---

## Estimated Time

2–3 days for implementation, testing, and documentation updates.

### PR DESCRIPTION
**Pull request by Anonymouseyy**
Fixes #4 

- Added wheel to requirements.txt for more straightforward building and development
- Implement StreamArray slicing, returns StreamArray object
- Updated docs to document slicing

### CODE REVIEW

**Comment by DannyAkintunde** [comment]
> Fixes #4 
> 
> - Added wheel to requirements.txt for more straightforward building and development
> - Implement StreamArray slicing, returns StreamArray object
> - Updated docs to document slicing

hmm, good work 😀

File: youtube_dl_scraper/core/stream_array.py [hunk 1]
```diff
@@ -241,18 +241,27 @@ def __len__(self) -> int:
...
 
-    def __getitem__(self, index: int) -> Stream:
+    def __getitem__(self, index: Union[int, slice]) -> Stream:
         """
-        Access a stream by its index.
+        Access a stream by its index or sliced stream array by its slice.
 
         Args:
-            index (int): The index of the stream.
+            inputs (Union[int, slice]):
```
Comment (DannyAkintunde) [inline youtube_dl_scraper/core/stream_array.py]: This should be `index` or you can rename the function argument from `index` -> `inputs`

**Review by DannyAkintunde — CHANGES_REQUESTED** [review]

File: youtube_dl_scraper/core/stream_array.py (hunk 1 above)
Comment (Anonymouseyy) [reply youtube_dl_scraper/core/stream_array.py]: gotcha
